src/Sarif.Driver/Sdk/ExportRulesMetadataCommandBase.cs

```diff
@@ -130,12 +130,16 @@ private void OutputSarifRulesMetada(string outputFilePath, ImmutableArray<IRule>
                 newRule.Id = rule.Id;
                 newRule.Name = rule.Name;
                 newRule.HelpUri = rule.HelpUri;
-                newRule.Properties = rule.Properties;
                 newRule.FullDescription = rule.FullDescription;
                 newRule.MessageFormats = rule.MessageFormats;
 
                 newRule.ShortDescription = rule.ShortDescription;
 
+                foreach (string propertyName in rule.PropertyNames)
+                {
+                    newRule.SetProperty(propertyName, rule.GetProperty(propertyName));
```

Comment:
A nice increase in usability.

---
In reply to: [62749682](https://github.com/Microsoft/sarif-sdk/pull/257#discussion_r62749682) [](ancestors = 62749682)